core/db/migrate/20130807024301_upgrade_adjustments.rb

```diff
@@ -14,7 +14,8 @@ def up
 
     # Tax adjustments have their sources altered
     Spree::Adjustment.where(:originator_type => "Spree::TaxRate").find_each do |adjustment|
-      adjustment.source = adjustment.originator
+      adjustment.source_id = adjustment.originator_id
+      adjustment.source_type = "Spree::TaxRate"
       adjustment.save
```

Comment:
@athal7 Worthy of a separate pull request to go along with the separate intentions so that one or the other can be reverted if necessary, but you're absolutely right.